components/forwarder/forwarder.go

```diff
@@ -0,0 +1,88 @@
+package forwarder
+
+import (
+	"context"
+
+	"github.com/ThreeDotsLabs/watermill"
+	"github.com/ThreeDotsLabs/watermill/message"
+	"github.com/pkg/errors"
+)
+
+// Forwarder subscribes to the topic provided in the config and publishes them to the destination topic embedded in the enveloped message.
+type Forwarder struct {
+	router    *message.Router
+	publisher message.Publisher
+	logger    watermill.LoggerAdapter
+	config    Config
+}
+
+// NewForwarder creates a forwarder which will subscribe to the topic provided in the provided config using the provided subscriber.
+// It will publish messages received on this subscription to the destination topic embedded in the enveloped message using the provided publisher.
+//
+// Provided subscriber and publisher can be from different Watermill Pub/Sub implementations, i.e. MySQL subscriber and Google Pub/Sub publisher.
+func NewForwarder(subscriberIn message.Subscriber, publisherOut message.Publisher, logger watermill.LoggerAdapter, config Config) (*Forwarder, error) {
+	config.setDefaults()
+
+	router, err := message.NewRouter(message.RouterConfig{}, logger)
+	if err != nil {
+		return nil, err
+	}
+
+	f := &Forwarder{router, publisherOut, logger, config}
+
+	router.AddNoPublisherHandler(
```

Comment:
I was thinking 'why not `AddHandler`, but then realized that the topic is dynamic and needs to be unraveled from the envelope.
I guess a publisher decorator could take care of that, but probably readability might suffer

Reply:
Yeah, I didn't think about using publisher decorator here, but I don't think it would give any additional value as the handler method is quite simple. I would opt for leaving it as it is for readability sake.

Reply:
👍